File: api/routes/usage.ts
/**
 * Usage route handler.
 * GET /api/usage — get usage stats
 * Extracted verbatim from api/[...path].ts — lines 1666-1686.
 */
import type { VercelRequest, VercelResponse } from '@vercel/node';
import { getSupabaseAdmin } from '../_lib/supabase';
import { ok, methodNotAllowed, requireAuth } from '../_lib/helpers';
import { getUserProfile } from '../_lib/billing-utils';

export default async function handleUsage(req: VercelRequest, res: VercelResponse, segments: string[]) {
  const supabase = getSupabaseAdmin();
  const user = await requireAuth(req, res);
  if (!user) return;
  if (req.method !== 'GET') return methodNotAllowed(res);

  const profile = await getUserProfile(supabase, user.id);
  if (!profile) return res.status(404).json({ error: 'Profile not found' });

  const plan = profile.subscription_plan || 'free';

  return ok(res, {
    plan,
    plan_label: plan.charAt(0).toUpperCase() + plan.slice(1),
    usage: {
      jobs: { used: profile.jobs_count || 0, limit: 999999, label: 'Job Roles' },
      assessments: { used: profile.assessments_count || 0, limit: 999999, label: 'Technical Assessments' },
      interviews: { used: profile.interviews_count || 0, limit: 999999, label: 'Interviews' },
    },
  });
}

File: backend/app/api/v2/endpoints/usage.py
from __future__ import annotations

from fastapi import APIRouter, Depends

from app.api.v2.deps import require_user
from app.auth.clerk import ClerkUser
from app.services.db.supabase_service import get_db_admin_service
from app.utils.responses import api_error, ok

router = APIRouter(prefix="/usage")


@router.get("")
async def get_usage(user: ClerkUser = Depends(require_user)):
    """Node-compatible: GET /api/usage

    Returns plan + usage counters from profiles.
    """

    db = get_db_admin_service()

    def _fetch_profile():
        return db.client.from_("profiles").select("*").eq("user_id", user.id).maybe_single().execute()

    res = await db.run(_fetch_profile)
    profile = getattr(res, "data", None)
    if not isinstance(profile, dict):
        return api_error(message="Profile not found", status_code=404)

    plan = str(profile.get("subscription_plan") or "free")
    plan_label = plan[:1].upper() + plan[1:]

    return ok(
        {
            "plan": plan,
            "plan_label": plan_label,
            "usage": {
                "jobs": {
                    "used": int(profile.get("jobs_count") or 0),
                    "limit": 999999,
                    "label": "Job Roles",
                },
                "assessments": {
                    "used": int(profile.get("assessments_count") or 0),
                    "limit": 999999,
                    "label": "Technical Assessments",
                },
                "interviews": {
                    "used": int(profile.get("interviews_count") or 0),
                    "limit": 999999,
                    "label": "Interviews",
                },
            },
        }
    )
